### ISSUE TITLE
Existing Drainage area for Large rivers

### ISSUE THREAD

**Issue by MatthewMeier** [Riverscapes/pyBRAT#19]
When you have a large river that passes through your watershed that Drainage area for upstream of where that river enters you watershed will appear as zero based on the relative window your looking at. A good way to solve this is to select the mainstem of that river that goes through your watershed and apply a DA value that you can find on Streamstats and then add it to the whole mainstem in your watershed.

**Comment by joewheaton** [comment]
Thanks @MatthewMeier for raising this issue that came up on Yakima. The issue is that we do not calculate a flow accumulation grid correctly unless the entire watershed is included as an input. When we break HUCs into their own processing units and we're dealing with a 'Middle' or a 'Lower', by definition a mainstem enters in at its boundary, and brings with it an upstream drainage area of the 'Upper' or 'Middle'.  So to get the correct drainage areas (and subsequent streampowers) for those mainstems, you need to add the upstream basins to the mainstems.  Lets discuss how to address this. I don't think this is an 'enhancement' yet as we have not made a request.

**Comment by bangen** [comment]
We don't run into a lot of basins where this occurs, but we had to address this issue in the Truckee watershed (which drains Lake Tahoe).  In the case of the Truckee there was a gage on the Truckee River at the outlet of Lake Tahoe.  I was able to use stream stats to get the upstream drainage area at the gage and then add the DA values to the Truckee River polylines.

In cases where a major river flows through a basin (i.e., doesn't start in a basin) the steps to updating the DA values should be:
- Get the upstream DA value either using a gage or delineate a basin using StreamStats at the location where the river crosses into the project HUC8
- Select all segments on that river and add the upstream DA value to the 'DA_sqkm' field

Note this should be done after running the 'BRAT Table' tool and before running the 'Braid Handler' tool (if using that tool) or 'iHyd' tool (if not using the 'Braid 'Handler' tool).

@MatthewMeier  I updated the 'segmentNetwork' script (commit 38b103fd5ae1246f07e919cfeba776c6a5c2f7f4) and it now carries over the 'GNIS_NAME' and populates it in a 'StreamName' field.  This makes it much easier to select the polylines that need the DA value updated.
